test/integration/oidc/oidc_test.go

```diff
@@ -0,0 +1,435 @@
+				t.Helper()
+
+				oidcServer.TokenHandler().EXPECT().Token().Times(1).DoAndReturn(oidcTest.DefaultTokenHandlerBehaviour(
+					t,
+					rsaPrivateKeyFilePath,
+					oidcServer.URL(),
+					defaultOIDCClientID,
+					defaultOIDCClaimedUsername,
+					defaultStubAccessToken,
+					defaultStubRefreshToken,
+					time.Now().Add(time.Second*1200).Unix(),
+				))
+				configureOIDCServerToReturnExpiredRefreshTokenErrorOnTryingToUpdateIDToken(oidcServer)
+			},
+			expectedErr: new(url.Error),
+		},
+		// This test case will be passed when PR
+		// https://github.com/kubernetes/kubernetes/pull/108794
+		// will be merged
+		/*{
+			name: "cache returns ok client if refresh token is updated in config",
+			configureUpdatingTokenBehaviour: func(t *testing.T, oidcServer *oidcTest.TestServer) {
+				t.Helper()
+				oidcServer.TokenHandler().EXPECT().Token().Times(1).DoAndReturn(oidcTest.DefaultTokenHandlerBehaviour(
+					t,
+					rsaPrivateKeyFilePath,
+					oidcServer.URL(),
+					defaultOIDCClientID,
+					defaultOIDCClaimedUsername,
+					defaultStubAccessToken,
+					"_new_fake_refresh_token_",
+					time.Now().Add(time.Second*1200).Unix(),
+				))
+			},
+			expectedErr: nil,
+		},*/
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			oidcServer, apiServer := configureTestInfrastructure(t)
+
+			expiredIDToken, stubRefreshToken := fetchExpiredToken(t, oidcServer)
+			configureClientConfigForOIDC(apiServer.ClientConfig, expiredIDToken, stubRefreshToken, oidcServer.URL())
+			expiredClient := kubernetes.NewForConfigOrDie(apiServer.ClientConfig)
+			configureOIDCServerToReturnExpiredRefreshTokenErrorOnTryingToUpdateIDToken(oidcServer)
+			_, err := expiredClient.CoreV1().Pods(defaultNamespace).List(context.Background(), metaV1.ListOptions{})
+			assert.Error(t, err)
+
+			tt.configureUpdatingTokenBehaviour(t, oidcServer)
+			idToken, stubRefreshToken := fetchOIDCCredentials(t, oidcServer.TokenURL(), caFilePath)
+			configureClientConfigForOIDC(apiServer.ClientConfig, idToken, stubRefreshToken, oidcServer.URL())
+			expectedOkClient := kubernetes.NewForConfigOrDie(apiServer.ClientConfig)
+			_, err = expectedOkClient.CoreV1().Pods(defaultNamespace).List(context.Background(), metaV1.ListOptions{})
+
+			if tt.expectedErr != nil {
+				assert.ErrorAs(t, err, &tt.expectedErr)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestClientSecret(t *testing.T) {
+	var tests = []struct {
+		name                              string
+		configureOIDCServerTokenBehaviour func(t *testing.T, oidcServer *oidcTest.TestServer)
+		expectedErrMessage                string
+	}{
+		{
+			name: "client secret is not passed",
+			configureOIDCServerTokenBehaviour: func(t *testing.T, oidcServer *oidcTest.TestServer) {
+				oidcServer.TokenHandler().EXPECT().Token().Times(2).Return(oidcTest.Token{}, oidcTest.ErrClientSecretNotPassed)
+			},
+			expectedErrMessage: oidcTest.ErrClientSecretNotPassed.Error(),
+		},
+		{
```

Comment:
you don't seem to be testing whether the client secret is actually posted to the oidc server, if it's in a correct format, and if it's the correct one